github-adapter/src/main/java/be/xplore/githubmetrics/githubadapter/config/GithubRestClientRequestObservationConvention.java

```diff
@@ -28,13 +28,9 @@ public KeyValues getLowCardinalityKeyValues(ClientRequestObservationContext cont
     }
 
     protected KeyValue uri(ClientRequestObservationContext context) {
-        String uri = Optional.ofNullable(context.getCarrier().getHeaders().get("path")).map(list -> {
-            if (!list.isEmpty()) {
-                return list.getFirst();
-            } else {
-                return "NO HEADER";
-            }
-        }).orElse("NO HEADER");
+        String uri = Optional.ofNullable(context.getUriTemplate())
+                .orElse("NONE");
+        uri = uri.split("\\?", 2)[0];
```

Comment:
you could put the split in a map before the `orElse` since it doesnt need to happen if the uri is None.
```java
Optional.ofNullable(context.getUriTemplate())
                .map(uri -> uri.split("\\?", 2)[0])
                .orElse("NONE");
```

Reply:
good idea. done. also means we can just return direct no need for variables